assets/javascripts/discourse/components/ai-persona-editor.gjs

```diff
@@ -0,0 +1,236 @@
+  @tracked showDelete = false;
+
+  @action
+  updateModel() {
+    this.editingModel = this.args.model.workingCopy();
+    this.showDelete = !this.args.model.isNew && !this.args.model.system;
+  }
+
+  @action
+  async updateAllGroups() {
+    this.allGroups = await Group.findAll();
+  }
+
+  @action
+  async save() {
+    const isNew = this.args.model.isNew;
+    this.isSaving = true;
+
+    const backupModel = this.args.model.workingCopy();
+
+    this.args.model.setProperties(this.editingModel);
+    try {
+      await this.args.model.save();
+      this.#sortPersonas();
+      if (isNew) {
+        this.args.personas.addObject(this.args.model);
+        this.router.transitionTo(
+          "adminPlugins.discourse-ai.ai-personas.show",
+          this.args.model
+        );
+      } else {
+        this.toasts.show({
+          data: { message: I18n.t("discourse_ai.ai-persona.saved") },
+          duration: 2000,
+        });
+      }
+    } catch (e) {
+      this.args.model.setProperties(backupModel);
+      popupAjaxError(e);
+    } finally {
+      later(() => {
+        this.isSaving = false;
+      }, 1000);
+    }
+  }
+
+  @action
+  delete() {
+    return this.dialog.confirm({
+      message: I18n.t("discourse_ai.ai-persona.confirm_delete"),
+      didConfirm: () => {
+        return this.args.model.destroyRecord().then(() => {
+          this.args.personas.removeObject(this.args.model);
+          this.router.transitionTo(
+            "adminPlugins.discourse-ai.ai-personas.index"
+          );
+        });
+      },
+    });
+  }
+
+  @action
+  updateAllowedGroups(ids) {
+    this.editingModel.set("allowed_group_ids", ids);
+  }
+
+  @action
+  async toggleEnabled() {
+    this.args.model.set("enabled", !this.args.model.enabled);
+    if (!this.args.model.isNew) {
+      try {
+        await this.args.model.update({ enabled: this.args.model.enabled });
+      } catch (e) {
+        popupAjaxError(e);
+      }
+    }
+  }
+
+  @action
+  async togglePriority() {
+    this.args.model.set("priority", !this.args.model.priority);
+    if (!this.args.model.isNew) {
+      try {
+        await this.args.model.update({ priority: this.args.model.priority });
+
+        this.#sortPersonas();
+      } catch (e) {
+        popupAjaxError(e);
+      }
+    }
+  }
+
+  @action
+  showPriorityHelp() {
+    this.dialog.alert({
+      message: I18n.t("discourse_ai.ai-persona.priority_help"),
+    });
+  }
+
+  #sortPersonas() {
+    const sorted = this.args.personas.toArray().sort((a, b) => {
+      if (a.priority && !b.priority) {
+        return -1;
+      } else if (!a.priority && b.priority) {
+        return 1;
+      } else {
+        return a.name.localeCompare(b.name);
+      }
+    });
+    this.args.personas.clear();
+    this.args.personas.setObjects(sorted);
+  }
+
+  <template>
+    <form
+      class="form-horizontal ai-persona-editor"
+      {{didUpdate this.updateModel @model.id}}
+      {{didInsert this.updateModel @model.id}}
```

Comment:
I don't think you need `@model.id` on either of these `didInsert` calls; that is used to tell `didUpdate` which property triggers the update

Reply:
I think I tried this and the component is reused when you navigate between personas so we need to issue it by observing the model id

Reply:
It's fine that the component is reused, it doesn't need to be destroyed + recreated every time, the model itself will update with `updateModel` because `@model.id` changed. Maybe `updateModel` isn't doing enough cleanup?